### PR TITLE
CentOS-6/Amazon Linux initscript

### PR DESCRIPTION
**Pull request by jnewbigin**
Fix .pid and lock file handling

### CODE REVIEW

File: packaging/linux/root/usr/share/buildkite-agent/lsb/buildkite-agent.sh [hunk 1]
```diff
@@ -15,10 +15,11 @@
 user="buildkite-agent"
 cmd="/usr/bin/buildkite-agent start"
 
-name=`basename $0`
-pid_file="/var/run/$name.pid"
-log="/var/log/$name.log"
-stderr_log="/var/log/$name.err"
+name=$(basename "$(readlink -f "$0")")
```
Comment (toolmantim) [inline packaging/linux/root/usr/share/buildkite-agent/lsb/buildkite-agent.sh]: Is there any reason we couldn't just hardcode "buildkite-agent" here?

**Comment by toolmantim** [comment]
Thanks so much! I'll merge this in for testing.

File: packaging/linux/root/usr/share/buildkite-agent/lsb/buildkite-agent.sh (hunk 1 above)
Comment (jnewbigin) [inline packaging/linux/root/usr/share/buildkite-agent/lsb/buildkite-agent.sh]: Using basename does allow you to run multiple instances with (a bit) less fuss:
cp /etc/rc.d/init.d/buildkite-agent  /etc/rc.d/init.d/buildkite-agent-2
chkconfig --add buildkite-agent-2
service buildkite-agent-2 start

**Comment by toolmantim** [comment]
Ok, I just tested and this works a treat. Thanks so much!

File: packaging/linux/root/usr/share/buildkite-agent/lsb/buildkite-agent.sh (hunk 1 above)
Comment (toolmantim) [inline packaging/linux/root/usr/share/buildkite-agent/lsb/buildkite-agent.sh]: Right right! Yeah… let's leave that in then :)

**Pull request merged**